kordac/processors/GenericTagBlockProcessor.py

```diff
@@ -0,0 +1,63 @@
+from markdown.blockprocessors import BlockProcessor
+from kordac.processors.utils import etree, parse_arguments, process_parameters
+import re
+
+
+class GenericTagBlockProcessor(BlockProcessor):
+    ''' A generic processor that matches '{<name> args}' and replaces
+    with the according html template.
+    '''
+    def __init__(self, processor, ext, *args, **kwargs):
+        '''
+        Args:
+            ext: An instance of the Kordac Extension.
+        '''
+        super().__init__(*args, **kwargs)
+        self.processor = processor
+        self.pattern = re.compile(r'(^|\n) *\{{{0} ?(?P<args>[^\}}]*)\}} *(\n|$)'.format(self.processor))
+        self.arguments = ext.processor_info[self.processor]['arguments']
+        template_name = ext.processor_info.get('template_name', self.processor)
+        self.template = ext.jinja_templates[template_name]
+        self.template_parameters = ext.processor_info[self.processor].get('template_parameters', None)
+        self.process_parameters = lambda processor, parameters, argument_values: \
+            process_parameters(ext, processor, parameters, argument_values)
+
+    def test(self, parent, block):
+        ''' Tests a block to see if the run method should be applied.
+
+        Args:
+            parent: The parent node of the element tree that children
+            will reside in.
+            block: The block to be tested.
+
+        Returns:
+            True if there is a match within the block.
+        '''
+        return self.pattern.search(block) is not None
+
+    def run(self, parent, blocks):
+        ''' Generic run method for single match tags.
+
+        Args:
+            parent: The parent node of the element tree that children
+            will reside in.
+            blocks: A list of strings of the document, where the
+            first block tests true.
```

Comment:
Same as earlier:
Should
> Returns:
   Markdown document with comments removed

be here too?...